addons/survey/views/survey_question_views.xml

```diff
@@ -202,8 +202,12 @@
                                     <field name="is_conditional" attrs="{'invisible': [('questions_selection', '=', 'random')]}"/>
                                     <field name="triggering_question_id" options="{'no_open': True, 'no_create': True}"
                                         attrs="{'invisible': [('is_conditional','=', False)], 'required': [('is_conditional','=', True)]}"/>
+                                    <field name="is_matrix_conditional" invisible="1"/>
+                                    <field name="triggering_row_id" options="{'no_open': True, 'no_create': True}"
+                                        attrs="{'invisible': ['|', '|', ('is_conditional','=', False), ('triggering_question_id','=', False), ('is_matrix_conditional','=', False)],
+                                                'required': ['&amp;', ('is_conditional','=', True), ('is_matrix_conditional','=', True)]}"/>
```

Comment:
as said, the is_matrix_conditional seems like a duplicated field.
is_matrix_conditional = is_conditional and question.type == 'matrix' and that could be done in view directly.
```
'invisible': ['|', '|', ('is_conditional','=', False), ('triggering_question_id','=', False), ('question_type', '!=', 'matrix')],
'required': [('is_conditional','=', True), ('question_type', '=', 'matrix')]
```

Reply:
Indeed, but the question type in the domain must be the one of the selected triggering question.
I therefore used a related field `triggering_question_type` to achieve this.